### PR TITLE
Restore log coverage along with misc niceties

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

https://github.com/anza-xyz/agave/pull/1260 temporarily disabled coverage for logging

#### Summary of Changes

Restore it along with some other (opinionated) niceties while we're at the file.

### CODE REVIEW

File: scripts/coverage.sh [hunk 1]
```diff
@@ -69,7 +72,6 @@ grcov_common_args=(
   --ignore upload-perf\*
   --ignore bench-streamer\*
   --ignore local-cluster\*
-  --ignore-not-existing
```
Comment (ryoqun) [inline scripts/coverage.sh:72]: let's see what happens. I'd like to avoid these squash-error flags if possible.

File: scripts/coverage.sh [hunk 2]
```diff
@@ -35,8 +35,6 @@ fi
 
 # Clean up
 rm -rf "$here/../target/cov/$COMMIT_HASH"
-find "$here/.." -type f -name '*.prof*' -exec rm {} +
-find "$here/.." -type f -name '*lcov*' -exec rm {} +
```
Comment (ryoqun) [inline scripts/coverage.sh:39]: i looks like we can avoid this intrusive `find`, thanks to proper ./target/cov use. (I have a bad habit of putting myriad of garbage files in my work tree for no good reason)

File: scripts/coverage.sh [hunk 3]
```diff
@@ -54,12 +52,17 @@ TEST_ARGS=(
   --skip shred::merkle::test::test_recover_merkle_shreds::
 )
 
-cargo +"$rust_nightly" test --target-dir "$here/../target/cov" "${PACKAGES[@]}" -- "${TEST_ARGS[@]}"
+# Most verbose log level (trace) is enabled for all solana code to make log!
+# macro code green always. Also, forcibly discard the vast amount of log by
+# redirecting the stderr altogether on CI, where all tests are run unlike
+# developing.
+RUST_LOG="solana=trace,agave=trace,$RUST_LOG" INTERCEPT_OUTPUT=/dev/null "$here/../ci/intercept.sh" \
```
Comment (ryoqun) [inline scripts/coverage.sh:59]: intercept.sh was handy for conditional discard depending on `$CI`.

File: scripts/coverage.sh [hunk 4]
```diff
@@ -54,12 +52,17 @@ TEST_ARGS=(
...
+  cargo +"$rust_nightly" test --target-dir "$here/../target/cov" "${PACKAGES[@]}" -- "${TEST_ARGS[@]}"
```
Comment (ryoqun) [inline scripts/coverage.sh:60]: hope no more overflow test failures in 1 hour.

Comment (ryoqun) [reply scripts/coverage.sh:60]: > hope no more overflow test failures in 1 hour.

seems we did it: https://buildkite.com/anza/agave/builds/4511#018f74f3-4da8-4cb2-aa63-d6c7d49d2f76

**Comment by ryoqun** [comment]
> ```diff
> @@            Coverage Diff            @@
> ##           master    #1328     +/-   ##
> =========================================
> + Coverage    81.6%    82.8%   +1.2%     
> =========================================
>   Files         867      867             
>   Lines      368900   368900             
> =========================================
> + Hits       301052   305592   +4540     
> + Misses      67848    63308   -4540     
> ```

hehe, coverage increased +1.2% treewide. :)

File: scripts/coverage.sh (hunk 1 above)
Comment (ryoqun) [reply scripts/coverage.sh:72]: @yihau seems this isn't needed for now from the fact the coverage step passed green? If you think this is still needed, could you share the reason?

Comment (yihau) [reply scripts/coverage.sh:72]: no. I think it is good to remove it if we can pass the test without it!

**Review by yihau — APPROVED** [review]
I'm still a little bit confused that why we need trace level log for CI env but I'm neutral about this one.

**Comment by ryoqun** [comment]
> I'm still a little bit confused that why we need trace level log for CI env but I'm neutral about this one.

Happy to elaborate a bit. :)

Firstly, this improves coverage like this:

image

In this way, it's convenient for human to spot any red lines in prs by eyes thanks to fewer redlines both when developing and code-reviewing.

Also, as we uncovered #1290, logging code isn't generally well tested. so, it's desirable for our ci to run the logging code as much as possible (currently indirectly; admittedly this is overloading the _coverage_ step, like the nightly-only frozen-abi tests). In this way, we can avoid finding these painful panics when debugging urgent issue in production with `RUST_LOG=trace`.

Lastly, to be explicit, i made trace log output not to be uploaded to buildkite, unlike the previous behavior.

**Comment by yihau** [comment]
ah yeah. you are right! thank you!

**Pull request merged**